google/cloud/internal/future_impl.h

```diff
@@ -279,6 +299,10 @@ class future_shared_state_base {
    * member variable and does not satisfy the shared state.
    */
   std::unique_ptr<continuation_base> continuation_;
+
+  // Allow users "cancel" the future with the given callback.
+  std::atomic<bool> cancelled_ = ATOMIC_VAR_INIT(false);
```

Comment:
Why need this be atomic?  Or, more generally, what are the concurrency requirements of `cancel()`?  Must callers serialize `cancel()` calls?  If so, atomicity doesn't seem to be required.  If not, atomicity doesn't seem sufficient.

Reply:
@devbww Good point. Maybe I can change it to `bool`. It is sufficient for my use case.